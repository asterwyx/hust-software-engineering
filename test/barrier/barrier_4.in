preset user AQSJ

preset credit A 100
preset credit Q 200
preset credit S 234
preset credit J 333

preset gift A barrier 2
preset gift Q barrier 1
preset gift S barrier 2
preset gift J barrier 3

preset userloc A 2 0
preset userloc Q 16 0
preset userloc S 22 0
preset userloc J 34 0

preset nextuser A#hhh

block -2# put barrier at 0
step 5 #A:2->7
n
block -2 #Q
step 0#Q:16->16
n
block 6#S put barrier at tool house
step 6 #S:22->28 through
1
f
block 1 #J
step 1#J:34->35
1

dump




