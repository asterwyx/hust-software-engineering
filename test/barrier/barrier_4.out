user AQSJ

fund A 10000
fund Q 10000
fund S 10000
fund J 12000

credit A 100
credit Q 200
credit S 184 #234-50
credit J 333

gift A barrier 2
gift Q barrier 1
gift S barrier 3
gift J barrier 3

userloc A 7 0
userloc Q 16 0
userloc S 28 0
userloc J 35 0

nextuser A